### PR TITLE
Implement getLibraryPath()

### PR DESCRIPTION
**Pull request by eile**

### CODE REVIEW

File: lunchbox/dso.h [hunk 1]
```diff
@@ -64,8 +64,8 @@ class DSO : public boost::noncopyable
      *         exported by the DSO.
      * @version 1.0
      */
-    LUNCHBOX_API
-    void* getFunctionPointer( const std::string& functionName ) const;
+    LUNCHBOX_API void* getFunctionPointer( const std::string& functionName )
+        const;
```
Comment (tribal-tec) [inline lunchbox/dso.h]: This formatting looks better? -1 for me, but your choice.

File: lunchbox/file.cpp [hunk 2]
```diff
@@ -135,4 +135,25 @@ std::string getExecutablePath()
...
+#elif __APPLE__
+    const boost::filesystem::path path( exePath );
+
+    // foo.app/Contents/MacOS/foo
+    if( boost::algorithm::ends_with( exePath, ".app/Contents/MacOS" ))
+        return path.parent_path().parent_path().parent_path().string() + "/lib";
+    return path.parent_path().string() + "/lib";
+#else
+    const boost::filesystem::path path( exePath );
+    return path.parent_path().string() + "/lib";
```
Comment (tribal-tec) [inline lunchbox/file.cpp]: lib64 is not used by some Linux's? RHEL at least puts python libs in there, not sure if that matters here

Comment (eile) [inline lunchbox/file.cpp]: We still have lib: https://github.com/Eyescale/CMake/blob/master/Common.cmake#L68 and
```
[bbplxviz06 ~ master]$ ls Buildyard/Build/install
bin  include  lib  ports  share
```

File: lunchbox/dso.h (hunk 1 above)
Comment (eile) [inline lunchbox/dso.h]: reverted.

**Pull request merged**